Mathlib/CategoryTheory/MorphismProperty/OverAdjunction.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.MorphismProperty.Comma
+import Mathlib.CategoryTheory.Adjunction.Over
+import Mathlib.CategoryTheory.MorphismProperty.Limits
+
+/-!
+# Adjunction of pushforward and pullback in `P.Over Q X`
+
+A morphism `f : X ⟶ Y` defines two functors:
+
+- `Over.map`: post-composition with `f`
+- `Over.pullback`: base-change along `f`
+
+These are adjoint under suitable assumptions on `P` and `Q`.
+
+-/
+
+namespace CategoryTheory.MorphismProperty
+
+open Limits
+
+variable {T : Type*} [Category T] (P Q : MorphismProperty T) [Q.IsMultiplicative]
+variable {X Y : T} (f : X ⟶ Y)
+
+section Map
+
+variable {P} [P.IsStableUnderComposition] (hPf : P f)
+
+variable {f}
+
+/-- If `P` is stable under composition and `f : X ⟶ Y` satisfies `P`,
+this is the functor `P.Over Q X ⥤ P.Over Q Y` given by composing with `f`. -/
+def Over.map : P.Over Q X ⥤ P.Over Q Y :=
```

Comment:
```suggestion
@[simps! obj_left obj_hom map_left]
def Over.map : P.Over Q X ⥤ P.Over Q Y :=
```

Reply:
It seems there is a linting issue with the generated `map_left`. Stating the lemmas separately as you did is a workaround, but the best approach would be the point out the problem on Zulip (tagging Floris who may understand better what is happening).

Reply:
After spending two hours on minimizing and debugging this, I found the issue: It is the same as this one: https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/!4.234183.20.60simps.60.20maximum.20recursion.20depth. I removed `@[simp]` from the two bad simp lemmas (that can be proven by `simp`, but don't seem to get flagged by `simpNF`) and now everything works.